sources/packages/api/src/esdc-integration/e-cert-full-time-integration/e-cert-full-time-integration.service.ts

```diff
@@ -211,12 +215,92 @@ export class ECertFullTimeIntegrationService extends SFTPIntegrationBase<void> {
+   * MSFT-Response folder
+   * Filename pattern is `PEDU.PBC.CERTSFB.yyyymmdd.001`
+   * @returns full file paths for all response files from the
+   * E-Cert response folder.
+   */
+  async getResponseFilesFullPath(): Promise<string[]> {
+    let filesToProcess: Client.FileInfo[];
+    const client = await this.getClient();
+    try {
+      filesToProcess = await client.list(
+        `${this.esdcConfig.ftpResponseFolder}`,
+        new RegExp(
+          `^${this.esdcConfig.environmentCode}EDU.PBC.CERTSFB.*\\.DAT`,
+          "i",
+        ),
+      );
+    } finally {
+      await SshService.closeQuietly(client);
+    }
+    return filesToProcess.map(
+      (file) => `${this.esdcConfig.ftpResponseFolder}/${file.name}`,
+    );
+  }
+
+  // Generate Record
+
   /**
    * Transform the text lines in parsed objects specific to the integration process.
-   * TODO: read the feedback file.
    * @param remoteFilePath full remote file path with file name.
+   * @returns Parsed records from the file.
    */
-  downloadResponseFile(remoteFilePath: string): Promise<void> {
-    throw new Error("To be implemented.");
+  async downloadResponseFile(
+    remoteFilePath: string,
+  ): Promise<ECertResponseRecord[]> {
+    const fileLines = await this.downloadResponseFileLines(remoteFilePath);
+    /**
+     * Read the first line to check if the header code is the expected one.
+     * and remove header.
+     */
+    const header = ECertFileHeader.createFromLine(fileLines.shift());
+    if (header.recordTypeCode !== RecordTypeCodes.ECertHeader) {
+      this.logger.error(
+        `The E-Cert file ${remoteFilePath} has an invalid transaction code on header: ${header.recordTypeCode}`,
+      );
+      // If the header is not the expected one.
+      throw new Error(
+        "The E-Cert file has an invalid transaction code on header",
+      );
+    }
+
+    /**
+     * Read the last line to check if the trailer code is the expected one
+     * and remove trailer line.
+     */
+    const trailer = ECertFileFooter.createFromLine(fileLines.pop());
+    if (trailer.recordTypeCode !== RecordTypeCodes.ECertFooter) {
+      this.logger.error(
+        `The E-Cert file ${remoteFilePath} has an invalid transaction code on trailer: ${trailer.recordTypeCode}`,
+      );
+      // If the trailer is not the expected one.
+      throw new Error(
+        "The E-Cert file has an invalid transaction code on trailer",
+      );
+    }
+
+    /**
+     * Check if the number of records match the trailer record count
+     * Here total record count is the total records rejected
+     */
+    if (trailer.recordCount !== fileLines.length) {
+      this.logger.error(
+        `The E-Cert file ${remoteFilePath} has invalid number of records, expected ${trailer.recordCount} but got ${fileLines.length}`,
+      );
+      // If the number of records does not match the trailer record count..
+      throw new Error("The E-Cert file has invalid number of records");
+    }
+
```

Comment:
I see the trailer records contains the totalSINHash, are we getting the SIN in our records? if yes we need to Check if the sum total hash of SIN in the records match the trailer SIN hash total. Else throw an error.

Reply:
no, we are not parsing the SIN number in E-cert Response file

Reply:
@ann-aot, just because we are not parsing does not mean that we cannot parse and check it.
I really do not see a strong reason to check the SIN hash, but if there is some policy or whatever we need to follow. I am not aware of it.

Reply:
@andrewsignori-aot what @ann-aot told to me was, we are not retrieving the SIN in the Detail records. SO this SIN hash in the trailer records cannot be verified.

Reply:
I believe, when I checked with @abschwenker, he asked me to take `document number `and `error codes`, if there is any policy or whatever, then I can double-check with @abschwenker on Monday

Reply:
as discussed added sinhas check